akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -773,11 +793,17 @@ private[akka] class DDataShard(
   }
 
   private def getState(): Unit = {
-    (0 until numberOfKeys).map { i =>
+    (0 until numberOfKeys).foreach { i =>
       replicator ! Get(stateKeys(i), readMajority, Some(i))
     }
   }
 
+  override protected[akka] def aroundReceive(rcv: Receive, msg: Any): Unit = {
+    super.aroundReceive(rcv, msg)
+    if (!waiting)
+      unstash() // unstash one message
```

Comment:
here is the optimization to unstash one instead of unstashAll

Reply:
what if we don't receive any messages after recoveryComplete? Could things remain in the stash for some time?

Reply:
`recoveryComplete` is triggered by a message, `GetSuccess`, and is setting `waiting=false` before reaching here